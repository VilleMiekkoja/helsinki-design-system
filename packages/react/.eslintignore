**/*.d.ts
webpack.config.js
.templates
lib/
