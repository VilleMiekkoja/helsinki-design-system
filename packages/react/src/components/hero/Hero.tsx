import React from 'react';

// import core base styles
import 'hds-core';
import styles from './Hero.module.scss';
import classNames from '../../utils/classNames';
import { useTheme } from '../../hooks/useTheme';
import { Koros, KorosProps, KorosShiftSpacer } from '../koros';

type HTMLElementAttributes = React.HtmlHTMLAttributes<HTMLDivElement>;
type ImgElementAttributes = React.ImgHTMLAttributes<HTMLImageElement>;
export type HeroProps = React.PropsWithChildren<
  HTMLElementAttributes & {
    /**
     * Should texts be centered. Use when there is no image.
     */
    centeredContent?: boolean;
    /**
     * Image source
     */
    imageSrc?: string;
    /**
     * Koros properties. Accepts also boolean "hide", which hides the koros.
     * @see KorosProps
     */
    koros?: Omit<KorosProps, 'rotate'> & {
      hide?: boolean;
    };
    /*
     * Css variables for theming the hero
     */
    theme?: HeroCustomTheme;
    /**
     * Defines the hero variant
     */
    variant?: 'imageLeft' | 'imageRight' | 'backgroundTop' | 'wideImage' | 'diagonalKoros' | 'textOnly';
  }
>;

export interface HeroCustomTheme {
  /**
   * Sets the background color. Is also koros color, if --koros-color is not set.
   * Default --color-white
   */
  '--background-color'?: string;
  /**
   * Text color
   * Default --color-black-90
   */
  '--color'?: string;
  /**
   * How background image is positioned if it overflows the container. Is set to css property "object-position"
   * Default bottom right
   */
  '--image-position'?: string;
  /**
   * Koros of the koros
   * Default --background-color
   */
  '--koros-color'?: string;
  /**
   * The 'backgroundTop' variant might need a custom color for the bottom/mobile koros.
   * Default none
   */
  '--bottom-koros-color'?: string;
  /**
   * The "diagonalKoros" variant might need koros position adjustment, if texts overflow.
   * This is an inset value with "top right bottom left"
   * Default "auto auto 10% -33%"
   */
  '--diagonal-koros-inset'?: string;
  /**
   * Custom horizontal padding for small screens
   * Default --spacing-layout-2-xs
   */
  '--horizontal-padding-small'?: string;
  /**
   * Custom horizontal padding for medium screens
   * Default --spacing-layout-xs
   */
  '--horizontal-padding-medium'?: string;
  /**
   * Custom horizontal padding for large screens
   * Default --spacing-layout-xs
   */
  '--horizontal-padding-large'?: string;
  /**
   * Custom horizontal padding for x-large screens
   * Default --spacing-layout-xs
   */
  '--horizontal-padding-x-large'?: string;
}

export const Hero = ({
  children,
  variant,
  imageSrc,
  centeredContent,
  theme,
  koros,
  ...elementAttributes
}: HeroProps) => {
  const editableTheme = { ...theme };
  if (!editableTheme['--koros-color']) {
    editableTheme['--koros-color'] = 'var(--background-color)';
  }
  const currentVariant: HeroProps['variant'] = variant || (imageSrc ? 'imageLeft' : 'textOnly');
  const customThemeClass = useTheme<HeroCustomTheme>(styles.hero, editableTheme);
  const korosStyle = { fill: 'var(--koros-color)' };
  const hideKoros = !!koros?.hide;
  const canKorosBeFlipped = koros?.flipHorizontal !== false;

  const heroElementAttributes: HTMLElementAttributes = {
    ...elementAttributes,
    className: classNames(
      styles.hero,
      customThemeClass,
      styles[currentVariant],
      (elementAttributes as HTMLElementAttributes).className,
    ),
  };

  const Image = () => {
    return (
      /* eslint-disable-next-line jsx-a11y/alt-text */
      <img className={styles.image} src={imageSrc} aria-hidden alt="" />
    );
  };

  const ImageAsBackground = (props?: Pick<ImgElementAttributes, 'className'>) => {
    return (
      <div className={classNames(styles.withBackgroundBackground, props && props.className)}>
        <Image />
      </div>
    );
  };

  const TwoColumsImage = () => {
    return (
      <div key={`imageContainer-${currentVariant}`} className={styles.twoColumnsImageContainer}>
        <Image />
      </div>
    );
  };

  const Content = () => {
    const classNameList = centeredContent ? classNames(styles.card, styles.centeredContent) : classNames(styles.card);
    return <div className={classNameList}>{children}</div>;
  };

  if (currentVariant === 'backgroundTop') {
    const TopOrBottomKoros = ({ top }: { top?: boolean }) => {
      const className = top ? styles.topKoros : styles.bottomKoros;
      const korosFillColor =
        !top && theme && theme['--bottom-koros-color'] ? theme['--bottom-koros-color'] : 'var(--koros-color)';
      return (
        <Koros
          {...koros}
          shift
          compact
          className={`${(koros && koros.className) || ''} ${className}`}
          style={{ fill: korosFillColor }}
        />
      );
    };

    return (
      <div {...heroElementAttributes}>
        <div className={styles.withBackgroundContainer}>
          <ImageAsBackground />
          <TopOrBottomKoros top />
          <div className={classNames(styles.content)}>
            <Content />
            <div className={styles.emptyColumn} />
          </div>
        </div>
        <TopOrBottomKoros />
      </div>
    );
  }

  if (currentVariant === 'diagonalKoros') {
    return (
      <div {...heroElementAttributes}>
        <div className={styles.diagonalKorosWithBackgroundContainer}>
          <div className={styles.content}>
            <div className={styles.contentColums}>
              <Content />
              <div className={styles.emptyColumn} />
            </div>
            <div className={styles.diagonalKorosMobileKoros}>
              <Koros {...koros} flipHorizontal shift compact style={korosStyle} />
            </div>
          </div>
          <Koros {...koros} className={styles.diagonalKorosAndBackground} style={korosStyle} />
          <ImageAsBackground className={styles.diagonalKorosBackgroundContainer} />
        </div>
      </div>
    );
  }

  const hasImage = !!imageSrc && currentVariant !== 'textOnly';
  const columnStyle = hasImage && currentVariant !== 'wideImage' ? styles.twoColumns : styles.singleColumn;
  return (
    <div {...heroElementAttributes}>
      <div className={styles.container}>
        <div key="content" className={classNames(styles.content, columnStyle)}>
          {hasImage && currentVariant === 'imageLeft' && <TwoColumsImage />}
          <Content />
          {hasImage && currentVariant === 'imageRight' && <TwoColumsImage />}
          {!hideKoros && !canKorosBeFlipped && <KorosShiftSpacer {...koros} />}
        </div>
      </div>
      {!hideKoros && <Koros {...koros} flipHorizontal={canKorosBeFlipped} style={korosStyle} />}
      {hasImage && (
        <div key="imageContainer" className={classNames(styles.imageContainer, styles.imageBelowKoros)}>
          <Image />
        </div>
      )}
    </div>
  );
};
