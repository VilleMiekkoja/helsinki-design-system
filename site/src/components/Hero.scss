@import "./utils.scss";

.hero-container {
  --koros-height: 145px;
  --hero-height: 520px;

  background-color: var(--color-coat-of-arms);
  box-sizing: border-box;
  position: relative;
}

.hero-bg-image {
  aspect-ratio: 16 / 9;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  margin-top: calc(var(--koros-height) / -2);

  @include x-large-and-above-screen {
    aspect-ratio: unset;
    height: 100%;
    margin-top: 0;
    left: calc((100% - var(--breakpoint-xl)) / 2);
    right: 0;
    position: absolute;
    top: 0;
  }
}

.hero-wrapper {
  margin: 0 auto;

  @include x-large-and-above-screen {
    max-width: var(--breakpoint-xl);
  }
}

.hero {
  color: var(--color-white);
  overflow: hidden;
  position: relative;

  @include x-large-and-above-screen {
    min-height: var(--hero-height);
  }
}

.hero-content {
  --triangle-top-corner-x: calc(100% + var(--spacing-layout-xl));
  --triangle-bottom-corner-x: calc(100% + var(--koros-height) + var(--spacing-layout-xl));

  background-color: var(--color-coat-of-arms);
  box-sizing: border-box;
  left: var(--hero-small-left);
  max-width: 60ch;
  padding: var(--spacing-m) var(--spacing-s);
  position: relative;
  z-index: 3;

  @include x-large-and-above-screen {
    clip-path: polygon(0 0, var(--triangle-top-corner-x) 0, 0 var(--triangle-bottom-corner-x), 0% 50%);
    left: auto;
    max-width: initial;
    padding: var(--spacing-l) var(--spacing-m);
    width: 67%;

    &:after {
      clear: both;
      content: "";
      display: table;
    }
  }
}

.hero-content-shape {
  @include x-large-and-above-screen {
    float: right;
    min-height: var(--hero-height);
    shape-outside: polygon(87% 0%, 100% 0%, 100% 100%, 37% 100%);
    width: 100%;
  }
}

.hero-title {
  font-size: var(--fontsize-heading-xxl);
  font-weight: 400;
  line-height: var(--lineheight-s);
  letter-spacing: -0.012em;
  margin: var(--spacing-m) 0 0 var(--spacing-m);

  @include x-large-and-above-screen {
    font-size: var(--fontsize-heading-xxl);
    margin: var(--spacing-xl) 0 0 0;
  }

  @include medium-screen {
    font-size: var(--fontsize-heading-xl);
    margin: var(--spacing-s) 0 0 var(--spacing-s);
  }

  @include small-screen {
    font-size: var(--fontsize-heading-l);
    margin: 0;
  }
}

.hero-text {
  margin: var(--spacing-s) 0 var(--spacing-l) var(--spacing-m);
  line-height: var(--lineheight-l);
  font-size: var(--fontsize-body-xl);

  @include x-large-and-above-screen {
    font-size: var(--fontsize-body-xl);
    line-height: var(--lineheight-l);
    margin: var(--spacing-m) 0 var(--spacing-xl) 0;
    max-width: 460px;
  }

  @include medium-screen {
    font-size: var(--fontsize-body-l);
    margin: var(--spacing-s) 0 var(--spacing-l) var(--spacing-s);
  }

  @include small-screen {
    font-size: var(--fontsize-body-l);
    margin: var(--spacing-xs) 0 var(--spacing-m) 0;
  }
}

.hero-koros {
  fill: var(--color-coat-of-arms);
  z-index: 2;
}

.hero-koros-horizontal {
  display: block;
  position: relative;
  top: -50px;

  @include x-large-and-above-screen {
    display: none;
  }
}

.hero-koros-rotated {
  display: none;

  @include x-large-and-above-screen {
    display: block;
    position: absolute;
    top: var(--koros-height);
    width: calc(100% + var(--koros-height)) !important;
  }
}

.front-page-hero-button {
  margin: 0 0 0 var(--spacing-m);

  @include x-large-and-above-screen {
    margin: 0;
  }

  @include medium-screen {
    margin: 0 0 0 var(--spacing-s);
  }

  @include small-screen {
    margin: 0;
  }
}
