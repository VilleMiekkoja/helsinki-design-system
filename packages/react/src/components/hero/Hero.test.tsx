import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import { Hero } from './Hero';
import { Button } from '../button';

describe('<Hero /> spec', () => {
  const imageSrc = 'http://image.com/image.jpg';
  const DefaultCardContent = () => {
    return (
      <>
        <h1>This is the heading</h1>
        <p>Nullam ut nunc consectetur, accumsan nunc sed, luctus nisl. Curabitur lacinia!</p>
        <Button variant="secondary" role="link">
          Click me
        </Button>
      </>
    );
  };

  const TextAndOrImage = ({ imagePosition }: { imagePosition?: 'left' | 'right' }) => {
    return (
      <Hero>
        {imagePosition === 'left' && <Hero.Image src={imageSrc} />}
        <Hero.Card>
          <DefaultCardContent />
        </Hero.Card>
        {imagePosition === 'right' && <Hero.Image src={imageSrc} />}
      </Hero>
    );
  };

  const WithBackgroundImage = ({ imagePosition }: { imagePosition: 'top' | 'right' }) => {
    return (
      <Hero>
        {imagePosition === 'top' && <Hero.BackgroundImage src={imageSrc} />}
        <Hero.Card>
          <DefaultCardContent />
        </Hero.Card>
        {imagePosition === 'right' && <Hero.BackgroundImage src={imageSrc} />}
      </Hero>
    );
  };

  const BottomWideImage = () => (
    <Hero>
      <Hero.Card>
        <DefaultCardContent />
      </Hero.Card>
      <Hero.WideImage src={imageSrc} />
    </Hero>
  );

  const componentVariations = [
    () => <WithBackgroundImage imagePosition="right" />,
    () => <WithBackgroundImage imagePosition="top" />,
    () => <TextAndOrImage imagePosition="left" />,
    () => <TextAndOrImage imagePosition="right" />,
    () => <TextAndOrImage />,
    () => <BottomWideImage />,
  ];

  it('renders the component', () => {
    componentVariations.forEach((Component) => {
      const { asFragment } = render(<Component />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  it('should not have basic accessibility issues', async () => {
    // cannot use forEach with async/await
    // eslint-disable-next-line no-restricted-syntax
    for (const Component of componentVariations) {
      const { container } = render(<Component />);
      // eslint-disable-next-line no-await-in-loop
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    }
  });
});
