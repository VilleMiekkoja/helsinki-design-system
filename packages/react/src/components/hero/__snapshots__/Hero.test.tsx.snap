// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> spec renders the component 1`] = `
<DocumentFragment>
  <div
    class="hero custom-theme-1 diagonalKoros"
  >
    <div
      class="diagonalKorosWithBackgroundContainer"
    >
      <div
        class="content"
      >
        <div
          class="contentColums"
        >
          <div
            class="card"
          >
            <h1>
              This is the heading
            </h1>
            <p>
              Nullam ut nunc consectetur, accumsan nunc sed, luctus nisl. Curabitur lacinia!
            </p>
            <button
              class="button secondary theme-default size-default"
              role="link"
              type="button"
            >
              <span
                class="label"
              >
                Click me
              </span>
            </button>
          </div>
          <div
            class="emptyColumn"
          />
        </div>
        <div
          class="diagonalKorosMobileKoros"
        >
          <div
            class="koros basic"
            style="fill: var(--koros-color); transform: scaleY(-1); margin-bottom: -14px; height: 14px; overflow: hidden;"
          >
            <svg
              aria-hidden="true"
              height="85"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <pattern
                  height="85"
                  id="koros_basic-2"
                  patternUnits="userSpaceOnUse"
                  width="106"
                  x="0"
                  y="0"
                >
                  <path
                    d="M0,800h20V0c-4.9,0-5,2.6-9.9,2.6S5,0,0,0V800z"
                    transform="scale(5.3)"
                  />
                </pattern>
              </defs>
              <rect
                fill="url(#koros_basic-2)"
                height="85"
                width="100%"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="koros basic diagonalKorosAndBackground"
        style="fill: var(--koros-color);"
      >
        <svg
          aria-hidden="true"
          height="85"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <pattern
              height="85"
              id="koros_basic-3"
              patternUnits="userSpaceOnUse"
              width="106"
              x="0"
              y="0"
            >
              <path
                d="M0,800h20V0c-4.9,0-5,2.6-9.9,2.6S5,0,0,0V800z"
                transform="scale(5.3)"
              />
            </pattern>
          </defs>
          <rect
            fill="url(#koros_basic-3)"
            height="85"
            width="100%"
          />
        </svg>
      </div>
      <div
        class="withBackgroundBackground diagonalKorosBackgroundContainer"
      >
        <img
          alt=""
          aria-hidden="true"
          class="image"
          src="http://image.com/image.jpg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Hero /> spec renders the component 2`] = `
<DocumentFragment>
  <div
    class="hero custom-theme-4 backgroundTop"
  >
    <div
      class="withBackgroundContainer"
    >
      <div
        class="withBackgroundBackground"
      >
        <img
          alt=""
          aria-hidden="true"
          class="image"
          src="http://image.com/image.jpg"
        />
      </div>
      <div
        class="koros basic  topKoros"
        style="fill: var(--koros-color); margin-top: -14px; height: 14px; overflow: hidden;"
      >
        <svg
          aria-hidden="true"
          height="85"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <pattern
              height="85"
              id="koros_basic-5"
              patternUnits="userSpaceOnUse"
              width="106"
              x="0"
              y="0"
            >
              <path
                d="M0,800h20V0c-4.9,0-5,2.6-9.9,2.6S5,0,0,0V800z"
                transform="scale(5.3)"
              />
            </pattern>
          </defs>
          <rect
            fill="url(#koros_basic-5)"
            height="85"
            width="100%"
          />
        </svg>
      </div>
      <div
        class="content"
      >
        <div
          class="card"
        >
          <h1>
            This is the heading
          </h1>
          <p>
            Nullam ut nunc consectetur, accumsan nunc sed, luctus nisl. Curabitur lacinia!
          </p>
          <button
            class="button secondary theme-default size-default"
            role="link"
            type="button"
          >
            <span
              class="label"
            >
              Click me
            </span>
          </button>
        </div>
        <div
          class="emptyColumn"
        />
      </div>
    </div>
    <div
      class="koros basic  bottomKoros"
      style="fill: var(--koros-color); margin-top: -14px; height: 14px; overflow: hidden;"
    >
      <svg
        aria-hidden="true"
        height="85"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <pattern
            height="85"
            id="koros_basic-6"
            patternUnits="userSpaceOnUse"
            width="106"
            x="0"
            y="0"
          >
            <path
              d="M0,800h20V0c-4.9,0-5,2.6-9.9,2.6S5,0,0,0V800z"
              transform="scale(5.3)"
            />
          </pattern>
        </defs>
        <rect
          fill="url(#koros_basic-6)"
          height="85"
          width="100%"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Hero /> spec renders the component 3`] = `
<DocumentFragment>
  <div
    class="hero custom-theme-7 imageLeft"
  >
    <div
      class="container"
    >
      <div
        class="content twoColumns"
      >
        <div
          class="twoColumnsImageContainer"
        >
          <img
            alt=""
            aria-hidden="true"
            class="image"
            src="http://image.com/image.jpg"
          />
        </div>
        <div
          class="card"
        >
          <h1>
            This is the heading
          </h1>
          <p>
            Nullam ut nunc consectetur, accumsan nunc sed, luctus nisl. Curabitur lacinia!
          </p>
          <button
            class="button secondary theme-default size-default"
            role="link"
            type="button"
          >
            <span
              class="label"
            >
              Click me
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="koros basic"
      style="fill: var(--koros-color); transform: scaleY(-1); margin-bottom: -14px; height: 14px; overflow: hidden;"
    >
      <svg
        aria-hidden="true"
        height="85"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <pattern
            height="85"
            id="koros_basic-8"
            patternUnits="userSpaceOnUse"
            width="106"
            x="0"
            y="0"
          >
            <path
              d="M0,800h20V0c-4.9,0-5,2.6-9.9,2.6S5,0,0,0V800z"
              transform="scale(5.3)"
            />
          </pattern>
        </defs>
        <rect
          fill="url(#koros_basic-8)"
          height="85"
          width="100%"
        />
      </svg>
    </div>
    <div
      class="imageContainer imageBelowKoros"
    >
      <img
        alt=""
        aria-hidden="true"
        class="image"
        src="http://image.com/image.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Hero /> spec renders the component 4`] = `
<DocumentFragment>
  <div
    class="hero custom-theme-9 imageRight"
  >
    <div
      class="container"
    >
      <div
        class="content twoColumns"
      >
        <div
          class="card"
        >
          <h1>
            This is the heading
          </h1>
          <p>
            Nullam ut nunc consectetur, accumsan nunc sed, luctus nisl. Curabitur lacinia!
          </p>
          <button
            class="button secondary theme-default size-default"
            role="link"
            type="button"
          >
            <span
              class="label"
            >
              Click me
            </span>
          </button>
        </div>
        <div
          class="twoColumnsImageContainer"
        >
          <img
            alt=""
            aria-hidden="true"
            class="image"
            src="http://image.com/image.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="koros basic"
      style="fill: var(--koros-color); transform: scaleY(-1); margin-bottom: -14px; height: 14px; overflow: hidden;"
    >
      <svg
        aria-hidden="true"
        height="85"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <pattern
            height="85"
            id="koros_basic-10"
            patternUnits="userSpaceOnUse"
            width="106"
            x="0"
            y="0"
          >
            <path
              d="M0,800h20V0c-4.9,0-5,2.6-9.9,2.6S5,0,0,0V800z"
              transform="scale(5.3)"
            />
          </pattern>
        </defs>
        <rect
          fill="url(#koros_basic-10)"
          height="85"
          width="100%"
        />
      </svg>
    </div>
    <div
      class="imageContainer imageBelowKoros"
    >
      <img
        alt=""
        aria-hidden="true"
        class="image"
        src="http://image.com/image.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Hero /> spec renders the component 5`] = `
<DocumentFragment>
  <div
    class="hero custom-theme-11 imageLeft"
  >
    <div
      class="container"
    >
      <div
        class="content twoColumns"
      >
        <div
          class="twoColumnsImageContainer"
        >
          <img
            alt=""
            aria-hidden="true"
            class="image"
            src="http://image.com/image.jpg"
          />
        </div>
        <div
          class="card"
        >
          <h1>
            This is the heading
          </h1>
          <p>
            Nullam ut nunc consectetur, accumsan nunc sed, luctus nisl. Curabitur lacinia!
          </p>
          <button
            class="button secondary theme-default size-default"
            role="link"
            type="button"
          >
            <span
              class="label"
            >
              Click me
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="koros basic"
      style="fill: var(--koros-color); transform: scaleY(-1); margin-bottom: -14px; height: 14px; overflow: hidden;"
    >
      <svg
        aria-hidden="true"
        height="85"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <pattern
            height="85"
            id="koros_basic-12"
            patternUnits="userSpaceOnUse"
            width="106"
            x="0"
            y="0"
          >
            <path
              d="M0,800h20V0c-4.9,0-5,2.6-9.9,2.6S5,0,0,0V800z"
              transform="scale(5.3)"
            />
          </pattern>
        </defs>
        <rect
          fill="url(#koros_basic-12)"
          height="85"
          width="100%"
        />
      </svg>
    </div>
    <div
      class="imageContainer imageBelowKoros"
    >
      <img
        alt=""
        aria-hidden="true"
        class="image"
        src="http://image.com/image.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Hero /> spec renders the component 6`] = `
<DocumentFragment>
  <div
    class="hero custom-theme-13 imageBottom"
  >
    <div
      class="container"
    >
      <div
        class="content singleColumn"
      >
        <div
          class="card"
        >
          <h1>
            This is the heading
          </h1>
          <p>
            Nullam ut nunc consectetur, accumsan nunc sed, luctus nisl. Curabitur lacinia!
          </p>
          <button
            class="button secondary theme-default size-default"
            role="link"
            type="button"
          >
            <span
              class="label"
            >
              Click me
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="koros basic"
      style="fill: var(--koros-color); transform: scaleY(-1); margin-bottom: -14px; height: 14px; overflow: hidden;"
    >
      <svg
        aria-hidden="true"
        height="85"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <pattern
            height="85"
            id="koros_basic-14"
            patternUnits="userSpaceOnUse"
            width="106"
            x="0"
            y="0"
          >
            <path
              d="M0,800h20V0c-4.9,0-5,2.6-9.9,2.6S5,0,0,0V800z"
              transform="scale(5.3)"
            />
          </pattern>
        </defs>
        <rect
          fill="url(#koros_basic-14)"
          height="85"
          width="100%"
        />
      </svg>
    </div>
    <div
      class="imageContainer imageBelowKoros"
    >
      <img
        alt=""
        aria-hidden="true"
        class="image"
        src="http://image.com/image.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
